cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java

```diff
@@ -314,12 +315,15 @@ public List<CloudResourceStatus> terminateDatabaseServer(AuthenticatedContext au
     }
 
     @Override
-    public List<CloudResourceStatus> update(AuthenticatedContext authenticatedContext, CloudStack stack, List<CloudResource> resources)
+    public List<CloudResourceStatus> update(AuthenticatedContext authenticatedContext, CloudStack stack, List<CloudResource> resources, UpdateType type)
             throws QuotaExceededException {
-        AzureClient client = authenticatedContext.getParameter(AzureClient.class);
-        AzureStackView azureStackView = azureStackViewProvider
-                .getAzureStack(new AzureCredentialView(authenticatedContext.getCloudCredential()), stack, client, authenticatedContext);
-        return azureVerticalScaleService.verticalScale(authenticatedContext, stack, resources, azureStackView, client);
+        if (type.equals(UpdateType.VERTICAL_SCALE)) {
+            AzureClient client = authenticatedContext.getParameter(AzureClient.class);
+            AzureStackView azureStackView = azureStackViewProvider
+                    .getAzureStack(new AzureCredentialView(authenticatedContext.getCloudCredential()), stack, client, authenticatedContext);
+            return azureVerticalScaleService.verticalScale(authenticatedContext, stack, resources, azureStackView, client);
+        }
+        return new ArrayList<>();
```

Comment:
`AzureVerticalScaleService#verticalScale` returns with an empty immutable list, while here we return with an array list. It would be nice to have these in sync. Also is it right to return with empty list after a vertical scale?